oak_channel/SPEC.md

```diff
@@ -1,39 +1,38 @@
-# Oak Baremetal Communication Protocol Specification
+# Oak Restricted Kernel Communication Protocol Specification
 
 This document serves as a specification of the protocol used for communication
-between the Untrusted Launcher and Trusted Runtime. The Trusted Runtime should
-be able to communicate with any implementation of the Untrusted Launcher that
-implements this specification.
+between the Untrusted Launcher (running on the host) and Enclave Binry built
+with the Restricted Kernel (running inside a guest VM).
 
-In the specification below the Trusted Runtime will henceforth be referred to
-the as the "service". The Untrusted Launcher will be referred to the as the
-"client".
+In the specification below the Enclave Binary will be referred to the as the
+"service". The Untrusted Launcher will be referred to the as the "client".
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
 interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
 
 ## Layer Defintions
 
-Conceptually, the Oak Baremetal Communication Protocol is made up of three
-layers. The [Method Layer](#method-layer) uses the Oak IDL to define the methods
-exposed by the service. It is responsible for encoding method parameters and
-return values. The [Message](#message-layer) takes encoded parameters / return
-values and wraps them as distinct messages that include metadata. The
-[Frame](#frame-layer) creates individual frames from messages, which are then
-sent over the transport channel.
+Conceptually, the Oak Restricted Kernel Communication Protocol is made up of
+three layers. The [Invocation Layer](#invocation-layer) uses protocol buffers
+(protobuf) to define the methods exposed by the service. It is responsible for
+encoding requests and responses that wrap input parameters and return values.
+The [Message Layer](#message-layer) provides a logical wrapper for the encoded
+requests and responses. The [Frame Layer](#frame-layer) converts messages into
+frames which are then sent over the transport channel.
 
-### Method Layer
+### Invocation Layer
 
-Methods define the functionality exposed by service. Each method has clearly
-defined parameters and return types. The Oak IDL is used to define which methods
-are available, their id, parameter and return types. The Oak IDL is also
-responsible for encoding method parameters and return values to bytes.
+Methods define the functionality exposed by service. Each method defines input
+parameters and return types. Protocol buffers (protobuf) is used to define which
+methods are available, the method ids, input parameters and return types. It
```

Comment:
It uses ... -> The service converts the requests and responses into bytes using standard protobuf encoding.